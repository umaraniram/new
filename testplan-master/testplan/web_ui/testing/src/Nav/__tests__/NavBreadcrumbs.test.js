import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import NavBreadcrumbs from '../NavBreadcrumbs';

function defaultProps() {
  const entry = {
    uid: '123',
    name: 'test',
    status: 'passed',
    category: 'testplan',
    counter: {
      passed: 0,
      failed: 0,
    },
  };
  return {
    entries: [entry],
    handleNavClick: jest.fn(),
  };
}

describe('NavBreadcrumbs', () => {
  const props = defaultProps();

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    props.handleNavClick.mockClear();
  });

  it('shallow renders the correct HTML structure', () => {
    const navBreadcrumbs = shallow(
        <NavBreadcrumbs {...props} />
    );
    expect(navBreadcrumbs).toMatchSnapshot();
  });

  it('calls handleNavClick when nav entry has been clicked', () => {
    const handleNavClick = props.handleNavClick;
    const navBreadcrumbs = shallow(
      <NavBreadcrumbs {...props} />
    );

    navBreadcrumbs.find('li').simulate('click');
    expect(handleNavClick.mock.calls.length).toEqual(1);
  });
});
