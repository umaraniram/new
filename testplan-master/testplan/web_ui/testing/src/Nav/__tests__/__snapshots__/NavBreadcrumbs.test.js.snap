// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBreadcrumbs shallow renders the correct HTML structure 1`] = `
<div
  className="navBreadcrumbs_koaten"
>
  <ul
    className="breadcrumbContainer_1qdzaym"
  >
    <li
      key="123"
      onClick={[Function]}
    >
      <div
        className="breadcrumbEntry_qbjkug-o_O-unselectable_16maxrd"
      >
        <NavEntry
          caseCountFailed={0}
          caseCountPassed={0}
          displayTime={false}
          executionTime={null}
          name="test"
          status="passed"
          type="testplan"
        />
      </div>
    </li>
  </ul>
</div>
`;
